src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+                InventoryUtils.quickMoveSlot(x, (tickDelay.value * 16).toLong())
+            }
+            //InventoryUtils.quickMoveSlot(1, (tickDelay.value * 16).toLong())
+        } else if (InventoryUtils.getSlots(0, 35, getIdFromBlock(mat)) == null) {
+            MessageSendHelper.sendChatMessage("$chatName No ${mat.localizedName} was found in inventory")
+            mc.getSoundHandler().playSound(PositionedSoundRecord.getRecord(SoundEvents.ENTITY_EXPERIENCE_ORB_PICKUP, 1.0f, 1.0f))
+            disable()
+            return false
+        }
+        InventoryUtils.swapSlotToItem(getIdFromBlock(mat))
+
+        val hitVec = Vec3d(neighbour).add(0.5, 0.5, 0.5).add(Vec3d(opposite.directionVec).scale(0.5))
+        val neighbourBlock = mc.world.getBlockState(neighbour).block
+
+        if (!isSneaking && BlockUtils.blackList.contains(neighbourBlock) || BlockUtils.shulkerList.contains(neighbourBlock)) {
+            mc.player.connection.sendPacket(CPacketEntityAction(mc.player, CPacketEntityAction.Action.START_SNEAKING))
+            isSneaking = true
+        }
+        if (rotate.value) {
+            BlockUtils.faceVectorPacketInstant(hitVec)
+        }
+
+        mc.playerController.processRightClickBlock(mc.player, mc.world, neighbour, opposite, hitVec, EnumHand.MAIN_HAND)
+        mc.player.swingArm(EnumHand.MAIN_HAND)
+        mc.rightClickDelayTimer = 4
+
+        val noBreakAnimation = ModuleManager.getModuleT(NoBreakAnimation::class.java)!!
+        if (noBreakAnimation.isEnabled) {
+            noBreakAnimation.resetMining()
+        }
+        return true
+    }
+
+    private fun getPlaceableSide(pos: BlockPos): EnumFacing? {
+        for (side in EnumFacing.values()) {
+            val neighbour = pos.offset(side)
+            if (!mc.world.getBlockState(neighbour).block.canCollideCheck(mc.world.getBlockState(neighbour), false)) {
+                continue
+            }
+            val blockState = mc.world.getBlockState(neighbour)
+            if (!blockState.material.isReplaceable) {
+                return side
+            }
+        }
+        return null
+    }
+
+    private fun lookAtBlock(pos: BlockPos) {
+        val vec3d = Vec3d(pos).add(0.5, 0.0, 0.5)
+        val lookAt = RotationUtils.getRotationTo(vec3d, true)
+        mc.player.rotationYaw = lookAt.x.toFloat()
+        mc.player.rotationPitch = lookAt.y.toFloat()
+    }
+
+    fun getPlayerDirection(): Int {
+        val yaw = (mc.player.rotationYaw % 360 + 360) % 360
+        return if (yaw >= 158 && yaw < 203) { 0 } //NORTH
+        else if (yaw >= 203 && yaw < 258) { 1 } //NORTH-EAST
+        else if (yaw >= 258 && yaw < 293) { 2 } //EAST
+        else if (yaw >= 293 && yaw < 338) { 3 } //SOUTH-EAST
+        else if (yaw >= 338 || yaw < 23) { 4 } //SOUTH
+        else if (yaw >= 23 && yaw < 68) { 5 } //SOUTH-WEST
+        else if (yaw >= 68 && yaw < 113) { 6 } //WEST
+        else { 7 } //NORTH-WEST
+    }
```

Comment:
This is already a util but this looks better. Mind modifying the existing on in MathsUtils?